src/SearchQueryService/Controllers/SearchController.cs

```diff
@@ -126,7 +128,52 @@ private static object CreateAzSearchResponse(AzPost newDocs)
         }
 
         private async Task PostDocuments(string indexName, AzPost docs)
-            => await _solrService.PostDocumentAsync(ConvertAzDocs(docs), indexName);
+        {
+            var delete = new AzPost { Value = new List<Dictionary<string, JsonElement>>() };
+            var addOrUpdate = new AzPost { Value = new List<Dictionary<string, JsonElement>>() };
+
+            foreach (var doc in docs.Value)
+            {
+                if (doc[_searchAction].GetString() == "delete")
+                {
+                    delete.Value.Add(doc);
+                }
+                else
+                {
+                    addOrUpdate.Value.Add(doc);
+                }
+            }
+
+            if (delete.Value.Count > 0)
+            {
+                await _solrService.DeleteDocumentsAsync(ConvertAzDocsForDelete(delete), indexName);
+            }
+
+            if (addOrUpdate.Value.Count > 0)
+            {
+                await _solrService.PostDocumentsAsync(ConvertAzDocs(addOrUpdate), indexName);
+            }
+        }
+
+        private static List<SolrDelete> ConvertAzDocsForDelete(AzPost azDocs)
+        {
+            var parsedDocs = new List<SolrDelete>();
+            foreach (var doc in azDocs.Value)
+            {
+                var convertedDocument = new SolrDelete();
+                foreach (var kv in doc)
```

Comment:
What if the document does not contain an Id? It would be better to know about it and throw the descriptive exception.
Btw, you can use `Linq` to find the `Id`.